### PR TITLE
Update links

### PR DESCRIPTION
**Pull request by XhmikosR**
* use https when possible
* replace coverage with a link to the docs page

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---

There are still plenty of 404s in our files, but I'll leave that for someone else

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,11 +66,11 @@ Sending a donation using our links below is **extremely helpful** in offsetting
...
 If you'd rather not [donate](https://pi-hole.net/donate/) (_which is okay!_), there are other ways you can help support us:
 - [Patreon](https://patreon.com/pihole) _Become a patron for rewards_
-- [Digital Ocean](http://www.digitalocean.com/?refcode=344d234950e1) _affiliate link_
+- [Digital Ocean](https://www.digitalocean.com/?refcode=344d234950e1) _affiliate link_
 - [Stickermule](https://www.stickermule.com/unlock?ref_id=9127301701&utm_medium=link&utm_source=invite) _earn a $10 credit after your first purchase_
-- [Amazon](http://www.amazon.com/exec/obidos/redirect-home/pihole09-20) _affiliate link_
+- [Amazon](https://www.amazon.com/exec/obidos/redirect-home/pihole09-20) _affiliate link_
 - [DNS Made Easy](https://cp.dnsmadeeasy.com/u/133706) _affiliate link_
-- [Vultr](http://www.vultr.com/?ref=7190426) _affiliate link_
+- [Vultr](https://www.vultr.com/?ref=7190426) _affiliate link_
```
Comment (dschaper) [inline README.md:73]: Can you remove this line completely? No one has ever signed up for Vultr with our afflink. We don't even use them anymore.

Comment (XhmikosR) [reply README.md:73]: Should we remove them from the docs repo also?

Comment (dschaper) [reply README.md:73]: Yes please.

Comment (XhmikosR) [reply README.md]: Maybe this is a good chance to streamline the links here and in the docs repo?

https://docs.pi-hole.net/#pi-hole-is-free-but-powered-by-your-support

Comment (dschaper) [reply README.md]: Sounds good.

File: README.md [hunk 2]
```diff
@@ -164,48 +163,24 @@ The API can be accessed via [`telnet`](https://github.com/pi-hole/FTL), the Web
...
-- [Scott Helme: Securing DNS across all of my devices with Pi-Hole + DNS-over-HTTPS + 1.1.1.1](https://scotthelme.co.uk/securing-dns-across-all-of-my-devices-with-pihole-dns-over-https-1-1-1-1/) (April, 2018)
-- [Scott Helme: Catching and dealing with naughty devices on my home network](https://scotthelme.co.uk/catching-naughty-devices-on-my-home-network/) (April, 2018)
-- [Bloomberg Business Week: Brotherhood of the Ad blockers](https://www.bloomberg.com/news/features/2018-05-10/inside-the-brotherhood-of-pi-hole-ad-blockers) (May, 2018)
-- [Software Engineering Daily: Interview with the creator of Pi-hole](https://softwareengineeringdaily.com/2018/05/29/pi-hole-ad-blocker-hardware-with-jacob-salmela/) (May, 2018)
-- [Raspberry Pi: Block ads at home using Pi-hole and a Raspberry Pi](https://www.raspberrypi.org/blog/pi-hole-raspberry-pi/) (July, 2018)
-- [Troy Hunt: Mmm... Pi-hole...](https://www.troyhunt.com/mmm-pi-hole/) (September, 2018)
-- [PEBKAK Podcast: Interview With Jacob Salmela](https://www.jerseystudios.net/2018/10/11/150-pi-hole/) (October, 2018)
+
+<https://docs.pi-hole.net/main/coverage/>
```
Comment (XhmikosR) [inline README.md:177]: Maybe I should make the link to the heading itself?

Comment (dschaper) [reply README.md:177]: I don't understand the question. Do you mean links to the sections like `https://docs.pi-hole.net/main/coverage/#youtubetwitvideo` ?

Comment (XhmikosR) [reply README.md:177]: No, I meant make the "Coverage" heading a link pointing to the docs coverage :)

```markdown
### [Coverage](https://docs.pi-hole.net/main/coverage/)
```

Comment (dschaper) [reply README.md:177]: Oh! Yeah, that sounds good.

Comment (XhmikosR) [reply README.md]: Done!

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/pi-hole-5-1-released/35577/1

**Pull request merged**